mods/persistence/game/machinery/_machines_base/stock_parts/_stock_parts.dm

```diff
@@ -6,7 +6,8 @@
 	. = ..()
 	if(persistent_id && istype(loc, /obj/machinery))
 		var/obj/machinery/M = loc
-		M.install_component(src, FALSE, TRUE)
+		if(!(src in M.component_parts))
```

Comment:
If you got runtimes from this, we're gonna need to be extra vigilant with components init..